grunt_tasks/build.js

```diff
@@ -14,301 +14,98 @@
+            watch: _.extend({}, isDev ? webpackDevConfig : webpackProdConfig, {
+                watch: true,
+                keepalive: true
+            })
         },
-        module: {
-            loaders: [
-                { // ES2015
-                    test: /\.js$/,
-                    loader: 'babel-loader',
-                    exclude: [paths.node_modules]
-                },
-                { // Stylus
-                    test: /\.styl$/,
-                    loaders: [
-                        ExtractTextPlugin.extract('style-loader'),
-                        'css-loader',
-                        {
-                            loader: 'stylus-loader',
-                            query: {
-                                'resolve url': true
-                            }
-                        }
-                    ]
-                },
-                { // CSS
-                    test: /\.css$/,
-                    loaders: [
-                        ExtractTextPlugin.extract('style-loader'),
-                        'css-loader'
-                    ]
-                },
-                { // Jade
-                    test: /\.jade$/,
-                    loader: 'jade-loader',
-                    query: {
-                        doctype: 'html' // see @girder/pull/1469
-                    }
-                },
-                { // PNG, JPEG
-                    test: /\.(png|jpg)$/,
-                    loaders: [urlLoader(), fileLoader()]
-                },
-                { // WOFF
-                    test: /\.woff(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'application/font-woff' }), fileLoader()]
-                },
-                { // WOFF2
-                    test: /\.woff2(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'application/font-woff2' }), fileLoader()]
-                },
-                { // TTF
-                    test: /\.ttf(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'application/octet-stream' }), fileLoader()]
-                },
-                { // EOT
-                    test: /\.eot(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [fileLoader()]
-                },
-                { // SVG
-                    test: /\.svg(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'image/svg+xml' }), fileLoader()]
+        // The grunt-contrib-watch task below is one option for dev, as described here:
+        // https://github.com/webpack/webpack-with-common-libs/blob/master/Gruntfile.js
+        // but it could also be done using the webpack dev server.
+        // It is however A LOT SLOWER than using the built-in watch options in grunt-webpack
+        watch: {
+            app: {
+                files: getWatchFiles(paths.web_src),
+                tasks: ['warnWatch', isDev ? 'webpack:dev' : 'webpack:prod'],
```

Comment:
We have a `npm run watch` command that will be our documented way to watch, so we can point that to the "proper" command to use and get rid of everything else.